lib/auth.ts

```diff
@@ -25,6 +32,15 @@ export function requireAuthentication(
       };
     }
 
+    if (!session.user?.admin) {
+      return {
+        redirect: {
+          destination: `/${context.locale}/welcome-bienvenue/`,
```

Comment:
Should this redirect to an error page, giving them appropriate information on why they didn't go to the page they wanted to see? Bouncing them over the welcome page may be confusing if they think they are legitimately supposed to be allowed to see the originally requested page.

Reply:
I'll add in a temporary error page for 'not authorized - must be admin'.